packages/react-charting/UnitTests/AreaChartUT.test.tsx

```diff
@@ -0,0 +1,683 @@
+      fireEvent.mouseOver(points[1]);
+      fireEvent.click(points[0]);
+      // nearest circle to highlight has index 6
+      expect(points[6].getAttribute('r')).toEqual('1');
+    },
+  );
+
+  testWithoutWait(
+    'Should return circleRadius if activePoint matches the circle',
+    AreaChart,
+    { data: chartData },
+    container => {
+      const points = getById(container, /circle/i);
+      expect(points).toHaveLength(10);
+      fireEvent.focus(points[0]);
+      expect(points[0].getAttribute('r')).toEqual('8');
+    },
+  );
+
+  testWithoutWait(
+    'Should return 0 if !isCircleClicked, (activePoint != circle) and (nearestCircleToHighlight != xDataPoint)',
+    AreaChart,
+    { data: chartData },
+    container => {
+      const points = getById(container, /circle/i);
+      expect(points).toHaveLength(10);
+      fireEvent.focus(points[0]);
+      expect(points[1].getAttribute('r')).toEqual('0');
+    },
+  );
+
+  testWithoutWait(
+    'Should return 0 if isCircleClicked is true but nearestCircleToHighlight does not match xDataPoint',
+    AreaChart,
+    { data: chartData },
+    container => {
+      const points = getById(container, /circle/i);
+      expect(points).toHaveLength(10);
+      fireEvent.click(points[0]);
+      fireEvent.mouseOver(points[1]);
+      expect(points[2].getAttribute('r')).toEqual('0');
+    },
+  );
+
+  testWithoutWait(
+    'Should return 0 if nearestCircleToHighlight does not match xDataPoint and activePoint is not set',
+    AreaChart,
+    { data: chartData },
+    container => {
+      const points = getById(container, /circle/i);
+      expect(points).toHaveLength(10);
+      fireEvent.mouseOver(points[1]);
+      expect(points[2].getAttribute('r')).toEqual('0');
+    },
+  );
+
+  testWithoutWait(
+    'Should return circleRadius if nearestCircleToHighlight matches xDataPoint and activePoint is not set',
+    AreaChart,
+    { data: chartData },
+    container => {
+      const points = getById(container, /circle/i);
+      expect(points).toHaveLength(10);
+      fireEvent.mouseOver(points[6]);
+      expect(points[6].getAttribute('r')).toEqual('8');
+    },
+  );
+
+  testWithoutWait(
+    'Should return circleRadius if nearestCircleToHighlight matches xDataPoint, activePoint matches another circle',
+    AreaChart,
+    { data: chartData },
+    container => {
+      const points = getById(container, /circle/i);
+      expect(points).toHaveLength(10);
+      fireEvent.focus(points[0]);
+      fireEvent.mouseOver(points[6]);
+      expect(points[6].getAttribute('r')).toEqual('8');
+    },
+  );
+});
+
+runTest('_addDefaultColors', () => {
+  testWithoutWait(
+    'Should return an array with the same length as the input array',
+    AreaChart,
+    { data: chartDataWithoutDefaultColor },
+    container => {
+      const points = getById(container, /circle/i);
+      expect(points).toHaveLength(10);
+      points.forEach(point => {
+        expect(point).toHaveAttribute('fill');
```

Comment:
match with the expected colors to be assigned here

Reply:
combining with the next test